Graphics/GraphicsEngineOpenGL/src/BufferGLImpl.cpp

```diff
@@ -225,12 +225,12 @@ void BufferGLImpl::UpdateData(GLContextState& CtxState, Uint64 Offset, Uint64 Si
 
     // We must unbind VAO because otherwise we will break the bindings
     constexpr bool ResetVAO = true;
```

Comment:
I believe VAO needs to be reset only for `ARRAY_BUFFER` and `ELEMENT_ARRAY_BUFFER`:
```cpp
const auto ResetVAO = m_BindTarget == GL_ARRAY_BUFFER || m_BindTarget == GL_ELEMENT_ARRAY_BUFFER;
```